src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Feedback/RazorFeedbackDiagnosticFileProvider.cs

```diff
@@ -0,0 +1,112 @@
+{
+    [Shared]
+    [Export(typeof(IFeedbackDiagnosticFileProvider))]
+    internal class RazorFeedbackDiagnosticFileProvider : IFeedbackDiagnosticFileProvider
+    {
+        private readonly FeedbackLogDirectoryProvider _feedbackLogDirectoryProvider;
+        private readonly JoinableTaskFactory _joinableTaskFactory;
+
+        [ImportingConstructor]
+        public RazorFeedbackDiagnosticFileProvider(
+            JoinableTaskContext joinableTaskContext,
+            FeedbackLogDirectoryProvider feedbackLogDirectoryProvider)
+        {
+            if (joinableTaskContext is null)
+            {
+                throw new ArgumentNullException(nameof(joinableTaskContext));
+            }
+
+            if (feedbackLogDirectoryProvider is null)
+            {
+                throw new ArgumentNullException(nameof(feedbackLogDirectoryProvider));
+            }
+
+            _feedbackLogDirectoryProvider = feedbackLogDirectoryProvider;
+            _joinableTaskFactory = joinableTaskContext.Factory;
+        }
+
+        public IReadOnlyCollection<string> GetFiles()
+        {
+            if (!_feedbackLogDirectoryProvider.DirectoryCreated)
+            {
+                // No one requested to create any feedback logs, no reason for us to provide any logs.
+                return Array.Empty<string>();
+            }
+
+            var timestamp = DateTime.Now.ToString("yyyy-MM-dd-HH-mm-ss", CultureInfo.InvariantCulture);
+            var zipFileName = $"RazorLogs_{timestamp}.zip";
+            var zipFilePath = Path.Combine(Path.GetTempPath(), zipFileName);
+            var logDirectory = _feedbackLogDirectoryProvider.GetDirectory();
+            if (!Directory.Exists(logDirectory))
+            {
+                // This should never be the case, just being extra defensive.
+                return Array.Empty<string>();
+            }
+
+            _joinableTaskFactory.RunAsync(() => ZipLogsAsync(logDirectory, zipFilePath));
+
+            return new[] { zipFilePath };
+        }
+
+        private static async Task ZipLogsAsync(string logs, string zipFilePath)
+        {
+            // Because we still have a handle to the log files, we can't zip them. So copy them and then zip the copy
+            var copiedLogs = CopyLogsToDirectory(logs);
+
+            // Start another task to zip this in the background.
+            await Task.Run(() => ZipFile.CreateFromDirectory(copiedLogs, zipFilePath));
+
+            try
+            {
+                if (Directory.Exists(copiedLogs))
+                {
+                    Directory.Delete(copiedLogs, true);
+                }
+            }
+            catch (Exception)
+            {
+                // Swallow any cleanup exceptions, we did our best.
+            }
+        }
+
+        private static string CopyLogsToDirectory(string logs)
+        {
+            var copyLocation = logs + DateTime.Now.ToString("yyyy-MM-dd-HH-mm-ss", CultureInfo.InvariantCulture);
```

Comment:
This is the tiniest nit-pick, but it might be nice if we turned these two timestamp creation spots into a function or if we extracted the string for it out into a constant.